scene/main/node.cpp

```diff
@@ -1435,6 +1435,13 @@ TypedArray<Node> Node::find_children(const String &p_pattern, const String &p_ty
 	return ret;
 }
 
+void Node::reparent(Node *p_parent, bool p_keep_global_transform) {
+	ERR_FAIL_NULL(p_parent);
+	ERR_FAIL_NULL_MSG(data.parent, "Node needs a parent to be reparented.");
```

Comment:
Another condition:
```
	ERR_FAIL_COND_MSG(p_parent == data.parent, "Can't reparent Node to its current parent.");
```

Reply:
Doing this is technically correct, but ok.

Reply:
Hmm, maybe it should a no-op instead of error? :thinking:

Reply:
Makes sense.